upload/fileUpload.go

```diff
@@ -71,14 +73,37 @@ func getGooglePhotosAlbumID(name string, c *gphotos.Client) string {
 	return album.Id
 }
 
-func (f *Item) upload(completedUploads *completeduploads.Service) error {
+func (f *Item) upload(completedUploads *completeduploads.Service, uploadURLsService *uploadurls.Service) error {
 	albumID := getGooglePhotosAlbumID(f.album, f.client)
 	log.Printf("Uploading object: file=%s", f.path)
 
+	// check upload URL db for previous uploads
+	log.Println("Looking up upload URLs database for ", f.path)
+	curUploadURL, err := uploadURLsService.GetUploadURL(f.path)
+	if err != nil {
+		// Not found, not an error, just an empty upload URL
+		log.Println(err)
+	}
+
 	// upload the file content to Google Photos
-	_, err := f.client.UploadFile(f.path, albumID)
+	ptrUploadURL := &curUploadURL
+	_, err = f.client.UploadFileResumable(f.path, ptrUploadURL, albumID)
+	if err != nil {
+		err = errors.Annotate(err, "failed uploading image")
+	}
+
+	if err != nil && *ptrUploadURL != "" {
+		log.Printf("Error uploading file '%s', storing upload URL '%s'\n", f.path, *ptrUploadURL)
```

Comment:
You don't need `\n` ending. `log` put it as default.

Reply:
Indeed, done.